dotnet/packages/Microsoft.TeamsAI/Microsoft.TeamsAI/AI/Models/IPromptCompletionModelEvents.cs

```diff
@@ -0,0 +1,170 @@
+﻿using Microsoft.Bot.Builder;
+using Microsoft.Teams.AI.AI.Prompts;
+using Microsoft.Teams.AI.AI.Tokenizers;
+using Microsoft.Teams.AI.Application;
+using Microsoft.Teams.AI.State;
+
+namespace Microsoft.Teams.AI.AI.Models
+{
+    /// <summary>
+    /// Events emitted by a IPromptCompletionStreamingModel.
+    /// </summary>
+    public interface IPromptCompletionModelEvents
+    {
+
+        /// <summary>
+        /// Defines the method that is triggered before the model is called to complete a prompt.
+        /// </summary>
+        /// <returns></returns>
+        public delegate void BeforeCompletionHandler(object sender, BeforeCompletionEventArgs args);
```

Comment:
Is there a reason `BeforeCompletionEventArgs` expects a cancellation token? It should be passed only if the delegate is an "async" delegate ... i.e it returns a `Task` object.

Also the convention is to pass the cancellation token as the last parameter of the delegate. Here's an example: https://github.com/microsoft/teams-ai/blob/e4a556b65892920c171fdbccdb6a8ce7a895c596/dotnet/packages/Microsoft.TeamsAI/Microsoft.TeamsAI/Application/AdaptiveCards/AdaptiveCardsHandlers.cs#L74

Reply:
Discussed and removed, forgot to delete